soh/soh/Enhancements/randomizer/randomizer.cpp

```diff
@@ -2774,19 +2777,36 @@ void DrawRandoEditor(bool& open) {
         return;
     }
 
-    DrawPresetSelector(PRESET_TYPE_RANDOMIZER);
-
     bool disableEditingRandoSettings = CVar_GetS32("gRandoGenerating", 0) || CVar_GetS32("gOnFileSelectNameEntry", 0);
     ImGui::PushItemFlag(ImGuiItemFlags_Disabled, disableEditingRandoSettings);
     ImGui::PushStyleVar(ImGuiStyleVar_Alpha, ImGui::GetStyle().Alpha * (disableEditingRandoSettings ? 0.5f : 1.0f));
 
-    ImGui::Dummy(ImVec2(0.0f, 0.0f));
-    if (ImGui::Button("Generate Seed")) {
+    DrawPresetSelector(PRESET_TYPE_RANDOMIZER);
+
+    UIWidgets::Spacer(0);
+
+    ImGui::Text("Seed");
+    ImGui::InputText("##RandomizerSeed", seedInputBuffer, MAX_SEED_BUFFER_SIZE, ImGuiInputTextFlags_None);
```

Comment:
Seeds are numbers only, so we definitely need to restrict input to only numbers.  At minimum, this means that instead of using `ImGuiInputTextFlags_None`, you need to use `ImGuiInputTextFlags_CharsDecimal || ImGuiInputTextFlags_NoBlank`, but I'd also recommend adding `ImGuiInputTextFlags_CallbackCharFilter` and the associated callback as well to throw out the non-digit numbers that the first flag allows, which according to https://github.com/ocornut/imgui/blob/master/imgui.h are .+-*/ (that may mean NoBlank isn't necessary).

Reply:
My thought was to support text based seeds (like `blargh`). since it's already supported on the 3ds side where this seed gets piped into. I guess the question becomes, are we fine with supporting text based seeds, or do we want to restrict to just numbers?

Reply:
There's no technical reason not to at the moment, but I don't like it since non-user-generated seeds are strictly numbers, and for user-generated ones they actually turn it into a number under the hood (if you look at the seed parameter in Playthrough_Init, it's just an int thanks to callers of that function hashing whatever the seed was), which adds obfuscation over what the text seed will _actually_ turn into.  Might be worth a discussion on Discord though.  

And either way, it's probably still a good idea to filter out non-alphanumeric characters for security purposes.

Reply:
I don't have a strong preference here, but using any text as a seed is also supported on n64 rando

Reply:
I guess to clarify my obfuscation comment, I worry that it's possible for non-numerical seeds to alias numerical seeds without knowing how C++'s hash function is operating.  We already ran into this before where we added the settings into the hash (see conversation on #1424).

Reply:
I update this input field to be numerical only and limited to uint32 max value.